components/raftstore/src/store/snap.rs

```diff
@@ -1453,8 +1453,18 @@ impl SnapManagerCore {
     fn get_total_snap_size(&self) -> Result<u64> {
         let mut total_size = 0;
         for entry in fs::read_dir(&self.base)? {
-            let entry = entry?;
-            let metadata = entry.metadata()?;
+            macro_rules! ignore_not_found {
+                ($op:expr) => {
+                    match $op {
+                        Err(e) if e.kind() == ErrorKind::NotFound => continue,
```

Comment:
what's the case that the entry is not found.

Reply:
`read_dir` and read entries is not atomic. It's possible that snapshots are deleted between them, in which case we can meet error `no such file or directory`.